### ISSUE TITLE
v2.8.0-pre fails when using persistent SSH connections with password authentication

### ISSUE THREAD

**Issue by slonopotamus** [git-lfs/git-lfs#3719]
```
$ git --version
git version 2.22.0
$ git lfs --version
git-lfs/2.8.0 (GitHub; linux amd64; go 1.12.7; git 928f7fe7)
$ uname -r
4.19.52-gentoo
```

Way to reproduce:

1. Set up persistent SSH connections (a.k.a. SSH multiplexing) in `~/.ssh/config`:

```
Host 172.105.67.160
    ControlMaster auto
    ControlPath ~/.ssh/controlmasters/%r@%h:%p
    ControlPersist 10m
```

2. Add git-lfs binary from #3718 on `$PATH`

3. `GIT_TRACE=1 git clone git@172.105.67.160:testrepo.git`
ssh password is "SecretPassword" without quotes.

Expected: clone completes successfully.

Actual:
```
$ GIT_TRACE=1 git clone git@172.105.67.160:testrepo.git 
14:05:25.335324 git.c:439               trace: built-in: git clone git@172.105.67.160:testrepo.git
Cloning into 'testrepo'...
14:05:25.336679 run-command.c:663       trace: run_command: unset GIT_DIR; ssh git@172.105.67.160 'git-upload-pack '\''testrepo.git'\'''
14:05:25.523641 run-command.c:663       trace: run_command: git index-pack --stdin -v --fix-thin '--keep=fetch-pack 8151 on radchenko-d-lnx' --check-self-contained-and-connected
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (4/4), done.
14:05:25.545374 git.c:439               trace: built-in: git index-pack --stdin -v --fix-thin '--keep=fetch-pack 8151 on radchenko-d-lnx' --check-self-contained-and-connected
Receiving objects: 100% (6/6), done.
remote: Total 6 (delta 0), reused 0 (delta 0)
14:05:25.576473 run-command.c:663       trace: run_command: git rev-list --objects --stdin --not --all --quiet '--progress=Checking connectivity'
14:05:25.595442 git.c:439               trace: built-in: git rev-list --objects --stdin --not --all --quiet '--progress=Checking connectivity'
14:05:25.596947 run-command.c:663       trace: run_command: 'git-lfs filter-process'
14:05:25.606251 trace git-lfs: exec: git 'version'
14:05:25.608986 trace git-lfs: exec: git 'config' '-l'
14:05:25.610437 trace git-lfs: Install hook: pre-push, force=false, path=/home/marat/testrepo/.git/hooks/pre-push
14:05:25.610502 trace git-lfs: Install hook: post-checkout, force=false, path=/home/marat/testrepo/.git/hooks/post-checkout
14:05:25.610532 trace git-lfs: Install hook: post-commit, force=false, path=/home/marat/testrepo/.git/hooks/post-commit
14:05:25.610558 trace git-lfs: Install hook: post-merge, force=false, path=/home/marat/testrepo/.git/hooks/post-merge
14:05:25.610581 trace git-lfs: Initialize filter-process
14:05:25.610835 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
14:05:25.613055 trace git-lfs: tq: running as batched queue, batch size of 100
14:05:25.613218 trace git-lfs: filepathfilter: accepting "1.bin"
14:05:25.613466 trace git-lfs: tq: sending batch of size 1
14:05:25.613625 trace git-lfs: run_command: ssh -- git@172.105.67.160 git-lfs-authenticate testrepo.git download
14:05:25.865063 trace git-lfs: api: batch 1 files
14:05:25.865416 trace git-lfs: HTTP: POST http://172.105.67.160:8123/testrepo.git/info/lfs/objects/batch
14:05:25.963361 trace git-lfs: HTTP: 200
14:05:25.963388 trace git-lfs: HTTP: {
  "objects" : [ {
    "oid" : "b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c",
    "size" : 4,
    "actions" : {
      "download" : {
        "href" : "http://172.105.67.160:8123/testrepo.git/info/lfs/storage/b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c",
        "header" : {
          "Authorization" : "Bearer eyJhbGciOiJBMTI4S1ciLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0.je1eU08T_Oa2ztzfNfFt6Oh5DtFlr1BqPTmRaJhFi-qYsrkvqaXppQ.3c1rMIDAwW_-GQc_BJi7bQ.ljbVC226UgG8-6QztzHE0b62gjim
14:05:25.963429 trace git-lfs: HTTP: ewo5WB_Z-I24yGAX3Nl93pR20VNgqB5qnMYOjrllRvPIMkZgyYRNMTIZOw4-pXYlQyycBo8TUkA5ZPt4bzbuRh-6piIZD_18O7YCjVwrm3CauQMc_pTIodrIAgRU5h7d7_yC6T_Ef7b08S6u8daUqofZC6ThpDpA74Umm4aAZvQxBanczcfiji1aZg.wFCbht08Mg3hPzG2GsytKg"
        }
      }
    }
  } ]
}
14:05:25.963500 trace git-lfs: tq: starting transfer adapter "basic"
14:05:25.963837 trace git-lfs: HTTP: GET http://172.105.67.160:8123/testrepo.git/info/lfs/storage/b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c
14:05:26.013770 trace git-lfs: HTTP: 200
read /dev/stdin: resource temporarily unavailable
error: external filter 'git-lfs filter-process' failed
fatal: 1.bin: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry the checkout with 'git checkout -f HEAD'
```

Bug does **not** reproduce with git-lfs 2.7.2:

```
$ GIT_TRACE=1 git clone git@172.105.67.160:testrepo.git
14:06:27.205340 git.c:439               trace: built-in: git clone git@172.105.67.160:testrepo.git
Cloning into 'testrepo'...
14:06:27.206529 run-command.c:663       trace: run_command: unset GIT_DIR; ssh git@172.105.67.160 'git-upload-pack '\''testrepo.git'\'''
14:06:27.393568 run-command.c:663       trace: run_command: git index-pack --stdin -v --fix-thin '--keep=fetch-pack 8601 on radchenko-d-lnx' --check-self-contained-and-connected
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (4/4), done.
14:06:27.415440 git.c:439               trace: built-in: git index-pack --stdin -v --fix-thin '--keep=fetch-pack 8601 on radchenko-d-lnx' --check-self-contained-and-connected
Receiving objects: 100% (6/6), done.
remote: Total 6 (delta 0), reused 0 (delta 0)
14:06:27.444311 run-command.c:663       trace: run_command: git rev-list --objects --stdin --not --all --quiet '--progress=Checking connectivity'
14:06:27.475340 git.c:439               trace: built-in: git rev-list --objects --stdin --not --all --quiet '--progress=Checking connectivity'
14:06:27.476707 run-command.c:663       trace: run_command: 'git-lfs filter-process'
14:06:27.511330 trace git-lfs: exec: git 'version'
14:06:27.564441 trace git-lfs: exec: git 'config' '-l'
14:06:27.585618 trace git-lfs: Install hook: pre-push, force=false, path=/home/marat/testrepo/.git/hooks/pre-push
14:06:27.585683 trace git-lfs: Install hook: post-checkout, force=false, path=/home/marat/testrepo/.git/hooks/post-checkout
14:06:27.585713 trace git-lfs: Install hook: post-commit, force=false, path=/home/marat/testrepo/.git/hooks/post-commit
14:06:27.585742 trace git-lfs: Install hook: post-merge, force=false, path=/home/marat/testrepo/.git/hooks/post-merge
14:06:27.585764 trace git-lfs: Initialize filter-process
14:06:27.585887 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
14:06:27.605847 trace git-lfs: tq: running as batched queue, batch size of 100
14:06:27.606001 trace git-lfs: filepathfilter: accepting "1.bin"
14:06:27.606099 trace git-lfs: tq: sending batch of size 1
14:06:27.606208 trace git-lfs: run_command: ssh -- git@172.105.67.160 git-lfs-authenticate testrepo.git download
14:06:27.872796 trace git-lfs: api: batch 1 files
14:06:27.873149 trace git-lfs: HTTP: POST http://172.105.67.160:8123/testrepo.git/info/lfs/objects/batch
14:06:27.971331 trace git-lfs: HTTP: 200
14:06:27.971358 trace git-lfs: HTTP: {
  "objects" : [ {
    "oid" : "b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c",
    "size" : 4,
    "actions" : {
      "download" : {
        "href" : "http://172.105.67.160:8123/testrepo.git/info/lfs/storage/b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c",
        "header" : {
          "Authorization" : "Bearer eyJhbGciOiJBMTI4S1ciLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0.9xjcFEtVy9-nqrEOYdj2yKvYkY5EM_aMEHsavsGSwnCMcHji_I9YCQ.TswYW0rQCXB7WiEmDhSZyQ.PA3oULDEOc28xx9xQbufypiEoSN0
14:06:27.971395 trace git-lfs: HTTP: PWwtnBYbbCKOYOecz7hiwzSpRRcHc7Yqi24JHizepHdqykSIwO-R5R1AlRPEXnXcRN-pO9GaDfondGcAsHfBtU7xnBHOrVxrDLPemYSIenJOTdqKIE-qkKoS9iBhcAgPXFYKO19FnVu6tO_1n9L4g4ixa9lHRr3DXc36tKfjPCoOawzKO8VJ0GNPjw.K98B2JkEF3-Hk4b_X7C3bg"
        }
      }
    }
  } ]
}
14:06:27.971467 trace git-lfs: tq: starting transfer adapter "basic"
14:06:27.971792 trace git-lfs: HTTP: GET http://172.105.67.160:8123/testrepo.git/info/lfs/storage/b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c
14:06:28.021782 trace git-lfs: HTTP: 200
14:06:28.022068 trace git-lfs: filepathfilter: accepting "1.bin"
14:06:28.022323 run-command.c:663       trace: run_command: /home/marat/testrepo/.git/hooks/post-checkout 0000000000000000000000000000000000000000 8871e15187d5c84c1f2946788deaea401ba0e51c 1
14:06:28.065342 git.c:704               trace: exec: git-lfs post-checkout 0000000000000000000000000000000000000000 8871e15187d5c84c1f2946788deaea401ba0e51c 1
14:06:28.065390 run-command.c:663       trace: run_command: git-lfs post-checkout 0000000000000000000000000000000000000000 8871e15187d5c84c1f2946788deaea401ba0e51c 1
14:06:28.125993 trace git-lfs: exec: git 'version'
14:06:28.175471 trace git-lfs: exec: git 'config' '-l'
14:06:28.191171 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
14:06:28.215682 trace git-lfs: filepathfilter: rewrite ".git" as "**/.git/**"
14:06:28.215701 trace git-lfs: filepathfilter: rewrite "**/.git" as "**/.git"
14:06:28.215730 trace git-lfs: filepathfilter: accepting "testrepo"
14:06:28.215782 trace git-lfs: filepathfilter: accepting "1.bin"
14:06:28.215789 trace git-lfs: filepathfilter: accepting ".gitattributes"
14:06:28.215794 trace git-lfs: filepathfilter: rejecting ".git" via ".git"
14:06:28.216538 run-command.c:46        trace: run_command: running exit handler for pid 8607
14:06:28.216631 trace git-lfs: filepathfilter: rewrite ".git" as "**/.git/**"
14:06:28.216647 trace git-lfs: filepathfilter: rewrite "**/.git" as "**/.git"
14:06:28.216688 trace git-lfs: filepathfilter: accepting "tmp"
```

Bug does **not** reproduce without persistent SSH connection.

Please note that this was set up in a quick and kinda insecure way. If by the time you want to test the issue again you find out that server needs to be reset to initial state, feel free to ping me.

**Comment by slonopotamus** [comment]
Okay, I shut down test server since root cause of this issue was found and going to be fixed in #3720

**Comment by bk2204** [comment]
That's fine. Thanks for providing the test server.
